pool/test/integration_utils.ml

```diff
@@ -0,0 +1,98 @@
+open Test_utils
+
+module AssignmentRepo = struct
+  let create session contact =
+    Assignment.(Created { contact; session_id = session.Session.id })
+    |> Pool_event.assignment
+    |> Pool_event.handle_event Data.database_label
+  ;;
+end
+
+module ContactRepo = struct
+  let create ?id ?(with_terms_accepted = false) () =
+    let open Utils.Lwt_result.Infix in
+    let contact = Model.create_contact ?id ~with_terms_accepted () in
+    let verified =
+      if contact.Contact.user.Sihl_user.confirmed
+      then Contact.[ Verified contact ]
+      else []
+    in
+    let%lwt () =
+      [ Contact.(
+          Created
+            { user_id = Contact.id contact
+            ; email = Contact.email_address contact
+            ; password =
+                contact.Contact.user.Sihl_user.password
+                |> Pool_user.Password.create
+                |> get_or_failwith_pool_error
+            ; firstname = Contact.firstname contact
+            ; lastname = Contact.lastname contact
+            ; terms_accepted_at = None
+            ; language = contact.language
+            })
+      ]
+      @ verified
+      |> Lwt_list.iter_s (Contact.handle_event Data.database_label)
+    in
+    contact
+    |> Contact.id
```

Comment:
How about opening `Contact` in this function, it's used many times.

Reply:
Cannot be opened before `let contact = Model.create_contact ?id ~with_terms_accepted () in` (Line 14) due to naming conflicts.
But yes